pkg/kubectl/resource_printer.go

```diff
@@ -353,103 +352,63 @@ func podHostString(host, ip string) string {
-		// a container is stuck in a crash loop.
-		message := getTermMsg(status.LastTerminationState.Terminated)
-		if message != "" {
-			message = "last termination: " + message
-		}
-		stateMsg := "Running"
-		if !status.Ready {
-			stateMsg = stateMsg + " *not ready*"
-		}
-		return stateMsg, translateTimestamp(state.Running.StartedAt), message, nil
-	} else if state.Terminated != nil {
-		return "Terminated", translateTimestamp(state.Terminated.StartedAt), getTermMsg(state.Terminated), nil
-	}
-	return "", "", "", fmt.Errorf("unknown container state %#v", *state)
+	shortHumanDuration := func(d time.Duration) string {
+		if seconds := int(d.Seconds()); seconds < 1 {
+			return "<1s"
+		} else if seconds < 60 {
+			return fmt.Sprintf("%ds", seconds)
+		} else if minutes := int(d.Minutes()); minutes == 1 {
+			return "~1m"
+		} else if minutes < 60 {
+			return fmt.Sprintf("%dm", minutes)
+		} else if hours := int(d.Hours()); hours == 1 {
+			return "~1h"
+		} else if hours < 48 {
+			return fmt.Sprintf("%dh", hours)
+		} else if hours < 24*7*2 {
+			return fmt.Sprintf("%dd", hours/24)
+		} else if hours < 24*30*3 {
+			return fmt.Sprintf("%dw", hours/24/7)
+		} else if hours < 24*365*2 {
+			return fmt.Sprintf("%dm", hours/24/30)
+		}
+		return fmt.Sprintf("%fy", d.Hours()/24/365)
+	}
+
+	return shortHumanDuration(time.Now().Sub(timestamp.Time))
 }
 
 func printPod(pod *api.Pod, w io.Writer, withNamespace bool) error {
-	var name string
+	name := pod.Name
 	if withNamespace {
-		name = types.NamespacedName{pod.Namespace, pod.Name}.String()
-	} else {
-		name = pod.Name
+		name = types.NamespacedName{Namespace: pod.Namespace, Name: pod.Name}.String()
 	}
 
-	_, err := fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n",
-		name,
-		pod.Status.PodIP,
-		"", "",
-		podHostString(pod.Spec.NodeName, pod.Status.HostIP),
-		formatLabels(pod.Labels),
-		pod.Status.Phase,
-		translateTimestamp(pod.CreationTimestamp),
-		pod.Status.Message,
-	)
-	if err != nil {
-		return err
-	}
-	// Lay out all containers on separate lines.
-	statuses := pod.Status.ContainerStatuses
-	if len(statuses) == 0 {
-		// Container status has not been reported yet. Print basic information
-		// of the containers and exit the function.
-		for _, container := range pod.Spec.Containers {
-			_, err := fmt.Fprintf(w, "%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n",
-				"", "", container.Name, container.Image, "", "", "", "", "")
-			if err != nil {
-				return err
-			}
+	restarts := 0
+	totalContainers := len(pod.Spec.Containers)
+	readyContainers := 0
+	reason := string(pod.Status.Phase)
+
+	for _, container := range pod.Status.ContainerStatuses {
+		restarts += container.RestartCount
+		if container.Ready {
+			readyContainers++
+		} else if container.State.Terminated != nil && container.State.Terminated.Reason != "" {
+			reason = container.State.Terminated.Reason
+		} else if container.State.Waiting != nil {
+			reason = container.State.Waiting.Reason
 		}
```

Comment:
I see the point, so I will add the pod message if there is any until we add the camelCase Reason to the `pod.Status`.

By the way I also noticed [this line](https://github.com/GoogleCloudPlatform/kubernetes/blob/d0daabb34bb87f4184d25b86ab39d04da3182161/plugin/pkg/scheduler/scheduler.go#L125). The messages from scheduler are not stored in `pod.Status.Message` is this a bug? Because it is the opposite from your example with the disk space error.